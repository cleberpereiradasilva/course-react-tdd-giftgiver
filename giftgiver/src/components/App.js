import React, { Component } from 'react'
import { Button } from 'react-bootstrap'
import Gift from './Gift'
import { max_number } from '../helper/index'

class App extends Component {
    constructor(props) {
        super(props)
        this.addGift = this.addGift.bind(this)        
        this.state = {
            gifts: []
        }
    }

   

    addGift() {
        const gifts = this.state.gifts
        const max_ids = gifts.map(gift => gift.id)
        const max_id = max_number(max_ids)
        gifts.push({ id: max_id + 1 })
        this.setState({
            gifts
        })
    }

    removeGift = id => {
        const gifts = this.state.gifts.filter(gift => gift.id !== id)
        this.setState({
            gifts
        })
    }
    
    
    render() {
        return (
            <div>
                <h1>Gifits</h1>                
                <div className='gift-list'>
                {
                    this.state.gifts.map(item => {
                        return (
                            <Gift key={item.id} 
                            gift={item.id}
                            handleRemove={this.removeGift}
                            />
                        )
                    })
                }
                </div>
                <Button className='btn-add btn btn-success' onClick={this.addGift}>
                    New Gift
                </Button>
            </div>
        )
    }
}

export default App
