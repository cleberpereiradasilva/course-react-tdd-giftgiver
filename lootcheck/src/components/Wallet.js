import React, { Component } from 'react'
import { connect } from 'react-redux'
import { deposit, withdraw } from '../actions/balance'


import { 
    Form, 
    FormGroup, 
    FormControl,     
    Col,
    Row,
    Button 
} from 'react-bootstrap'

export class Wallet extends Component{
    constructor(){
        super()
        this.state = {
            balance: undefined
        }
    }
    
    updateWallet = event => {
        this.setState({
            balance: parseInt(event.target.value, 10)
        })
    }
    deposit = () => this.props.deposit(this.state.balance)
    withdraw = () => this.props.withdraw(this.state.balance)
    
    render() {
        return (            
                <Row className='gift'>
                    <Form className='gift__form'>
                        <div className='row'>
                            <Col md={12}>
                                <h3 className='span-balance'>Wallet balance: {this.props.balance}</h3>
                            </Col>
                            <Col md={12}>
                            <FormGroup>                                
                                <FormControl className='input-wallet'                         
                                    onChange={this.updateWallet}  />
                            </FormGroup>
                            </Col>
                            <Col md={6} className='gift__form__row__button'>
                                <Button type='button' className='btn btn-success btn-deposit' 
                                onClick={this.deposit} >
                                    +
                                </Button>
                            </Col>
                            <Col md={6} className='gift__form__row__button'>
                                <Button type='button' className='btn btn-danger btn-withdraw' 
                                onClick={this.withdraw} >
                                    -
                                </Button>
                            </Col>
                            
                        </div>

                    </Form>
            </Row>
        )
    }
}

export default connect(state => { return { balance: state.balance }}, { deposit, withdraw })(Wallet)
