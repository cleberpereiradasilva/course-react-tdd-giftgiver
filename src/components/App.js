import React, { Component } from 'react'
import { Button } from 'react-bootstrap'

class App extends Component {
    constructor(props) {
        super(props)
        this.addGift = this.addGift.bind(this)
        this.state = {
            gifts: []
        }
    }

    addGift() {
        const gifts = this.state.gifts
        const max_ids = gifts.map(gift => gift.id)
        const max_id = max_ids.length == 0 ? 0 : Math.max(...max_ids)
        gifts.push({ id: max_id + 1 })
        this.setState({
            gifts
        })
    }
    
    render() {
        return (
            <div>
                <h1><Button className='btn-add' onClick={this.addGift}>Add Gifit</Button></h1>
            </div>
        )
    }
}

export default App
