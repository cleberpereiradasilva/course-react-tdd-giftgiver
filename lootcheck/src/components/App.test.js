import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import Wallet from './Wallet'

describe('App', ()=> {
    const app = shallow(<App />)
    

    it('app exist', ()=> {
        expect(app).toMatchSnapshot()
    })

    it('Wallet exist', ()=> {        
        expect(app.find('Connect(Wallet)').exists()).toBe(true)
    })
})
