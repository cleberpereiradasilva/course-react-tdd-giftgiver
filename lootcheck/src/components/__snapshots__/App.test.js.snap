// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app exist 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        Loot Check
    </h2>
</div>,
  "nodes": Array [
    <div>
      <h2>
            Loot Check
      </h2>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App />,
    "_forcedUpdate": false,
    "_instance": App {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <h2>
            Loot Check
      </h2>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
